nemo/collections/asr/parts/submodules/subsampling.py

```diff
@@ -264,7 +278,20 @@ def forward(self, x, lengths):
         )
         x = x.unsqueeze(1)
 
-        x = self.conv(x)
+        # split inputs if chunking_factor is set
+        # in dw conv, automatically avoiding a bug / feature limiting indexing of tensors to 2**31
+        # see https://github.com/pytorch/pytorch/issues/80020
+        # after the first conv numel will be * conv-channels but (self._stride * self._stride) downsampled
+        x_ceil = 2 ** 31 / self._conv_channels * self._stride * self._stride
+        if torch.numel(x) > x_ceil or self.chunking_factor != 1:
```

Comment:
Do the test for chunking factor first - numel costs non 0 time. 

Also do we want to auto switch to chunk mode ?

Reply:
the idea here to prevent certain failure if the chunking factor is set too low!

Reply:
To be clear, in this version, [automatic] chunking is always enabled if the 2^31 ceiling is reached, and the size of chunk is computed based on the 2^31 ceiling. If the user chooses to set the chunk size manually via the method, it will be used even if the ceiling is not reached. But, if the ceiling is reached -- in this implementation -- a [smaller] chunk size - compared to the one manually set -- is used to prevent failure.